Packages/Status/Sources/Status/Editor/StatusEditorViewModel.swift

```diff
@@ -126,6 +127,21 @@ public class StatusEditorViewModel: ObservableObject {
                          multiple: pollVotingFrequency.canVoteMultipleTimes,
                          expires_in: pollDuration.rawValue)
       }
+
+      if !hasExplicitlySelectedLanguage {
+          // Attempt language resolution using Natural Language
+          let recognizer = NLLanguageRecognizer()
+          recognizer.processString(statusText.string)
+          // Use languageHypotheses to get the probability with it
+          let hypotheses = recognizer.languageHypotheses(withMaximum: 1)
+          // Assert that 85% probability is enough :)
+          // A one word toot that is en/fr compatible is only ~50% confident, for instance
+          if let (language, probability) = hypotheses.first, probability > 0.85 {
```

Comment:
My only concern is that it can overwrite the user default setting. And also at the time of posting, if the language detected is different, then the UI is wrong.

Reply:
I can try to make the switch dynamically when the recognizer have enough probability? And the switch would be visible in the bar for the user. And with a little "pop" animation, the user see it happening?

I can manage to make the changes tonight

Reply:
But it means it would have to be run every time the user type. I'm more afraid of the performances issues then. I think it could be fine as is, let's merge and see the feedbacks.

Reply:
And/Or a switch within the settings ;)